plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -28,16 +28,17 @@ import {
 import { FormProps, IChangeEvent, withTheme } from '@rjsf/core';
 import { Theme as MuiTheme } from '@rjsf/material-ui';
 import React, { useState } from 'react';
+import { transformSchemaToProps } from './schema';
 
 const Form = withTheme(MuiTheme);
 type Step = {
-  schema: JSONSchema;
-  label: string;
+  schema: JsonObject;
```

Comment:
oh ok i see, you abandoned that one because we now have the ui things in there too - ok, so, future extension, make a type like that by using parts of JSONSchema and gluing them together with the extras

Reply:
Yep we opted for opaque data all the way from the entity schema to the frontend, where it is then unpacked. The coupling is between the template YAMLs and the implementation of the scaffolder frontend form rendering, nobody inbetween cares much about the schema format.

Reply:
kk